.circleci/config.yml

```diff
@@ -1,11 +1,100 @@
 version: 2.1
 orbs:
   node: circleci/node@4.0.0
+executors:
+  docker-executor:
+    # for docker you must specify an image to use for the primary container
+    docker:
+      - image: circleci/node:12.18
+        environment:
+          DATABASE_URL: postgresql://postgres@localhost/ttasmarthub
+      - image: circleci/postgres:12.4-ram
+        environment:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: secretpass
+          POSTGRES_DB: ttasmarthub
+commands:
+  create_combined_yarnlock:
+    description: "Concatenate all yarn.json files into single file.
+      File is used as checksum source for part of caching key."
+    parameters:
+      filename:
+        type: string
+        default: "combined-yarnlock.txt"
+    steps:
+      - run:
+          name: Combine package-lock.json files to single file
+          command: cat yarn.lock frontend/yarn.lock > << parameters.filename >>
 jobs:
   build:
-    docker:
-      - image: circleci/node:latest
+    executor: docker-executor
     steps:
       - checkout
+      - create_combined_yarnlock
+      - restore_cache:
+          keys:
+            # To manually bust the cache, increment the version numbers e.g. v2-yarn...
+            - v1-yarn-deps-{{ checksum "combined-yarnlock.txt" }}
+            # If checksum is new, restore partial cache
+            - v1-yarn-deps-
       - run: yarn deps
-      - run: yarn run test
+      - save_cache:
+          paths:
+            - combined-yarnlock.txt
```

Comment:
This will only cache the combined file, and not the two `node_modules` directories